subsys/logging/log_core.c

```diff
@@ -570,16 +571,23 @@ static void log_process_thread_func(void *dummy1, void *dummy2, void *dummy3)
 	}
 }
 
-K_THREAD_DEFINE(logging, CONFIG_LOG_PROCESS_THREAD_STACK_SIZE,
-		log_process_thread_func, NULL, NULL, NULL,
-		CONFIG_LOG_PROCESS_THREAD_PRIO, 0, K_NO_WAIT);
-#else
-#include <init.h>
+K_THREAD_STACK_DEFINE(logging_stack, CONFIG_LOG_PROCESS_THREAD_STACK_SIZE);
+struct k_thread logging_thread;
+#endif /* CONFIG_LOG_PROCESS_THREAD */
+
 static int enable_logger(struct device *arg)
 {
 	ARG_UNUSED(arg);
+#ifdef CONFIG_LOG_PROCESS_THREAD
+	/* start logging thread */
+	k_thread_create(&logging_thread, logging_stack,
+			K_THREAD_STACK_SIZEOF(logging_stack),
+			log_process_thread_func, NULL, NULL, NULL,
+			CONFIG_LOG_PROCESS_THREAD_PRIO, 0, K_NO_WAIT);
```

Comment:
please set the name of the thread. Previously it was set in the K_THREAD_DEFINE, when creating threads with k_thread_create this needs to be set with k_thread_name_set.